elfo-network/src/frame/read.rs

```diff
@@ -0,0 +1,235 @@
+use crate::{
+    codec::{
+        self,
+        decode::{DecodeState, DecodeStats},
+        format::NetworkEnvelope,
+    },
+    frame::{
+        buffers::{ReadBuffer, COMPRESSED_DATA_BUFFER_CAPACITY, DECOMPRESSED_DATA_BUFFER_CAPACITY},
+        lz4::{DecompressStats, LZ4Buffer},
+    },
+};
+
+use derive_more::Display;
+use eyre::{eyre, Result};
+
+#[derive(Display)]
```

Comment:
It's strange to implement `Display` for complex structures with logic, prefer printing `CompressionAlgorithm` instead